library/core/src/intrinsics/mod.rs

```diff
@@ -2042,6 +2046,18 @@ pub unsafe fn volatile_copy_memory<T>(_dst: *mut T, _src: *const T, _count: usiz
 /// The volatile parameter is set to `true`, so it will not be optimized out
 /// unless size is equal to zero.
 ///
+/// # Safety
```

Comment:
What would be much more important do clarify here is whether both the read and the write are volatile, or only one of them. Also, "the volatile parameter is set to `true`" makes no sense, that's talking about a parameter inside the codegen backend that shouldn't show up in such docs.